Mathlib/Algebra/BigOperators/Ring.lean

```diff
@@ -68,6 +68,11 @@ end Semiring
 
 section Semiring
 
+theorem dvd_sum [Semiring β]
```

Comment:
I'm guessing this change is to allow this lemma to be used for not-necessarily-commutative semirings, but do you know if it also works without assoc as well? ie, can this be moved below the NonAssocSemiring?

Reply:
Yes, I need to drop the redundant commutativity assumption.

Associativity cannot be dropped because the relevant `Dvd` instance [semigroupDvd](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Divisibility/Basic.html#semigroupDvd) requires it. (I also think this is appropriate, without divisibility without associativity seems quite a foreign concept.)

As a compromise I have added a commit which drops the unitality assumption.